src/org/labkey/test/components/ui/grids/CustomizeGridDialog.java

```diff
@@ -0,0 +1,491 @@
+        this.linkedComponent = linkedComponent;
+        waitForReady();
+    }
+
+    @Override
+    protected void waitForReady()
+    {
+        WebDriverWrapper.waitFor(()->
+                {
+                    WebElement spinner = BootstrapLocators.loadingSpinner.findWhenNeeded(this);
+                    return !spinner.isDisplayed() &&
+                            elementCache().contentPanelLocator.findElements(this).size() == 2;
+                },
+                "Customize Grid dialog did not render in time.", 1_500);
+    }
+
+    /**
+     * Check or uncheck the 'Show all system and user-defined fields' checkbox checked?
+     *
+     * @param checked Set to true to check the box, false to uncheck it.
+     * @return True if checked false otherwise.
+     */
+    public CustomizeGridDialog setShowAll(boolean checked)
+    {
+        if(checked)
+        {
+            elementCache().checkbox.check();
+        }
+        else
+        {
+            elementCache().checkbox.uncheck();
+        }
+
+        return this;
+    }
+
+    /**
+     * Is the 'Show all system and user-defined fields' checkbox checked?
+     *
+     * @return True if checked false otherwise.
+     */
+    public boolean isShowAllChecked()
+    {
+        return elementCache().checkbox.isChecked();
+    }
+
+    /**
+     * Get the list of visible fields from the 'Available Fields' panel. Children of an expanded filed only have the name
+     * in the list, this des not include the path.
+     *
+     * @return The list of field names.
+     */
+    public List<String> getAvailableFields()
+    {
+        List<WebElement> listItemElements = elementCache().getListItemNameElements(elementCache().availableFieldsPanel());
+        return listItemElements.stream().map(WebElement::getText).collect(Collectors.toList());
+    }
+
+    /**
+     * Add a top level field to the 'Shown in Grid' list.
+     *
+     * @param fieldName The name of the field.
+     * @return This dialog.
+     */
+    public CustomizeGridDialog addAvailableFieldToGrid(String fieldName)
+    {
+        WebElement listItem = elementCache().getListItemElementByFieldKey(elementCache().availableFieldsPanel(), fieldName);
+
+        Assert.assertTrue(String.format(FIELD_NOT_AVAILABLE, fieldName),
+                listItem.isDisplayed());
+
+        WebElement addIcon = Locator.tagWithClass("div", "view-field__action")
+                .withChild(Locator.tagWithClass("i", "fa-plus"))
+                        .findElement(listItem);
+
+        addIcon.click();
+
+        return this;
+    }
+
+    /**
+     * Add a nested field to the 'Shown in Grid' list. You must provide the list of parent fields.
+     *
+     * @param parentFields The list of parent fields to first expand.
+     * @param fieldName The field to add.
+     * @return This dialog.
+     */
+    public CustomizeGridDialog addAvailableFieldToGrid(List<String> parentFields, String fieldName)
```

Comment:
Combine both `addAvailableFieldToGrid` methods into one varargs method:
```suggestion
    public CustomizeGridDialog addAvailableFieldToGrid(String... fieldKeyParts)
```

Reply:
Changed.